dbt_style_guide.md

```diff
@@ -1,237 +1,415 @@
 
-select * from final
+- Prefer `union all` to `union` [*](http://docs.aws.amazon.com/redshift/latest/dg/c_example_unionall_query.html)
 
-```
+- Avoid table aliases in join conditions (especially initialisms) – it's harder to understand what the table called "c" is compared to "customers".
+
+- If joining two or more tables, _always_ prefix your column names with the table alias. If only selecting from one table, prefixes are not needed.
 
-- Your join should list the "left" table first (i.e. the table you are selecting `from`):
-```sql
-select
+- Be explicit about your join (i.e. write `inner join` instead of `join`). `left joins` are normally the most useful, `right joins` often indicate that you should change which table you select `from` and which one you `join` to.
+
+- When filtering by multiple clauses, each clause and expression should be on a new line.  
+Example:
+  ```sql
+  where 
+    user_id is not null
+    and status = 'pending'
+    and location = 'hq'
+  ```
+
+- Joins should list the "left" table first (i.e., the table you're joining data to):
+  ```sql
+  select
     trips.*,
     drivers.rating as driver_rating,
     riders.rating as rider_rating
-
-from trips
-left join users as drivers
+  from trips
+  left join users as drivers
     on trips.driver_id = drivers.user_id
-left join users as riders
+  left join users as riders
     on trips.rider_id = riders.user_id
 
-```
+  ```
+
+- ### Example SQL
+  ```sql
+  with
+
+  my_data as (
+      select * from {{ ref('my_data') }}
+      where not is_deleted
+  ),
+
+  some_cte as (
+      select * from {{ ref('some_cte') }}
+  ),
+
+  some_cte_agg as (
+      select
+          id,
+          sum(field_4) as total_field_4,
+          max(field_5) as max_field_5
+      from some_cte
+      group by 1
+  ),
+
+  final as (
+      select [distinct]
+          my_data.field_1,
+          my_data.field_2,
+          my_data.field_3,
+
+          -- use line breaks to visually separate calculations into blocks
+          case
+              when my_data.cancellation_date is null
+                and my_data.expiration_date is not null
+                then expiration_data
+              when my_data.cancellation_date is null
+                then my_data.start_date + 7
+              else my_data.cancellation_date
+          end as cancellation_date,
+
+          some_cte_agg.total_field_4,
+          some_cte_agg.max_field_5
+      from my_data
+      left join some_cte_agg  
+          on my_data.id = some_cte_agg.id
+      where 
+        my_data.field_1 = 'abc'
+        and (
+            my_data.field_2 = 'def'
+            or my_data.field_2 = 'ghi'
+        )
+      having count(*) > 1

```

Comment:
this is syntactically incorrect, you need a `group by` to use a `having`

Reply:
+1

Reply:
Good catch! How's [this instead](https://github.com/dbt-labs/corp/pull/63/commits/c25dbfe85ad4c5e5b31262683076345d043412d3)?